File: bj_9012.java
import java.util.*;
import java.io.*;

public class Main{
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int num = Integer.parseInt(br.readLine());
		String[] strs = new String[num];
		for(int i=0; i<num; i++) {
			strs[i] = br.readLine();
		}
		for(int i=0; i<num; i++) {
			Boolean isNo = false;
			Stack<Character> stack = new Stack<>();
			String str = strs[i];
			for(int j=0; j<str.length(); j++) {
				char c = str.charAt(j); 
				if(c == '(') stack.push(c);
				else {
					if(stack.isEmpty()) {
						System.out.println("NO");
						isNo = true;
						break;
					}
					else stack.pop();
				}
			}
			if(!isNo) {
				if(stack.isEmpty()) System.out.println("YES");
				else System.out.println("NO");
			}
		}
	}
}

File: bj_9012.kt
package com

import java.io.*
import java.util.*

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val num = br.readLine().toInt()
    val strs = arrayOfNulls<String>(num)
    for(i in 0..<num){
        strs[i] = br.readLine()
    }
    for(i in 0..<num){
        var isNo = false
        val stack = Stack<Char>()
        val str = strs[i]
        for(j in 0..<str!!.length){
            val c = str!!.get(j)
            if(c == '(') stack.push(c)
            else{
                if(stack.isEmpty()){
                    println("NO")
                    isNo = true
                    break
                }
                else stack.pop()
            }
        }
        if(!isNo){
            if(stack.isEmpty()) println("YES")
            else println("NO")
        }
    }
}